services/airtime/dtone/client_test.go

```diff
@@ -19,15 +19,15 @@ func TestClient(t *testing.T) {
 
 	mocks := httpx.NewMockRequestor(map[string][]httpx.MockResponse{
 		"https://airtime-api.dtone.com/cgi-bin/shop/topup": []httpx.MockResponse{
-			httpx.NewMockResponse(200, "info_txt=pong\r\n", nil),                  // successful ping
-			httpx.NewMockResponse(400, "error_code=1\r\nerror_txt=Oops\r\n", nil), // unsuccessful ping
-			httpx.NewMockResponse(200, withCRLF(msisdnResponse), nil),             // successful msdninfo query
-			httpx.NewMockResponse(200, "xxx=yyy\r\n", nil),                        // unexpected response to msdninfo query
-			httpx.NewMockResponse(200, withCRLF(reserveResponse), nil),            // successful reserve ID request
-			httpx.NewMockResponse(200, "xxx=yyy\r\n", nil),                        // unexpected response to reserve ID request
-			httpx.NewMockResponse(200, withCRLF(topupResponse), nil),              // successful topup request
-			httpx.NewMockResponse(200, "xxx=yyy\r\n", nil),                        // unexpected response to topup request
-			httpx.MockConnectionError,                                             // timeout
+			httpx.NewMockResponse(200, nil, "info_txt=pong\r\n", 1),                  // successful ping
```

Comment:
half of this diff is just adding a new param to mock responses to repeat the body - needed a JSON serializable way to make big payloads

Reply:
Seems having a different constructor may have made that easier no?

Reply:
yup!